components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/AdminRolePermissionsUpdateListener.java

```diff
@@ -0,0 +1,191 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.mgt.listener;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.CarbonConstants;
+import org.wso2.carbon.identity.api.resource.mgt.APIResourceMgtException;
+import org.wso2.carbon.identity.application.common.IdentityApplicationManagementException;
+import org.wso2.carbon.identity.application.common.model.AssociatedRolesConfig;
+import org.wso2.carbon.identity.application.common.model.Scope;
+import org.wso2.carbon.identity.application.common.model.ServiceProvider;
+import org.wso2.carbon.identity.application.mgt.ApplicationConstants;
+import org.wso2.carbon.identity.application.mgt.ApplicationManagementServiceImpl;
+import org.wso2.carbon.identity.application.mgt.internal.ApplicationManagementServiceComponentHolder;
+import org.wso2.carbon.identity.organization.management.service.exception.OrganizationManagementException;
+import org.wso2.carbon.identity.organization.management.service.util.OrganizationManagementUtil;
+import org.wso2.carbon.identity.role.v2.mgt.core.RoleManagementService;
+import org.wso2.carbon.identity.role.v2.mgt.core.exception.IdentityRoleManagementException;
+import org.wso2.carbon.identity.role.v2.mgt.core.model.Permission;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants.ORGANIZATION;
+
+/**
+ * Admin role permissions update listener to update admin role permissions.
+ */
+public class AdminRolePermissionsUpdateListener extends AbstractApplicationMgtListener  {
+
+    private static final Log LOG = LogFactory.getLog(AdminRolePermissionsUpdateListener.class);
+
+    @Override
+    public boolean doPostCreateApplication(ServiceProvider serviceProvider, String tenantDomain, String userName)
+            throws IdentityApplicationManagementException {
+
+        if (!isEnable()) {
+            LOG.debug("AdminRolePermissionUpdateListener is not enabled.");
+            return false;
+        }
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("AdminRolePermissionUpdateListener fired for tenant " +
+                    "creation for Tenant: " + tenantDomain);
+        }
+
+        if (!ApplicationConstants.CONSOLE_APPLICATION_NAME.equals(serviceProvider.getApplicationName())) {
+            return false;
+        }
+
+        try {
+            if (OrganizationManagementUtil.isOrganization(tenantDomain)) {
+                return false;
+            }
+            if (CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME) {
+                return false;
```

Comment:
same comment https://github.com/wso2/carbon-identity-framework/pull/5042/files#r1370707688 is applicable

Reply:
fixed